### PR TITLE
[NFC][PowerPC] Add getScalarIntVT to return MVT based on arch

### PR DESCRIPTION
**Pull request by lei137**
Add `getScalarIntVT()` to return scalar int VT based on if arch is 32 or 64bit.

### CODE REVIEW

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp [hunk 1]
```diff
@@ -5252,14 +5242,14 @@ static SDValue EmitTailCallStoreFPAndRetAddr(SelectionDAG &DAG, SDValue Chain,
...
-                     SmallVectorImpl<TailCallArgumentInfo>& TailCallArguments) {
+static void CalculateTailCallArgDest(
+    SelectionDAG &DAG, MachineFunction &MF, bool IsPPC64, SDValue Arg,
+    int SPDiff, unsigned ArgOffset,
+    SmallVectorImpl<TailCallArgumentInfo> &TailCallArguments) {
   int Offset = ArgOffset + SPDiff;
   uint32_t OpSize = (Arg.getValueSizeInBits() + 7) / 8;
   int FI = MF.getFrameInfo().CreateFixedObject(OpSize, Offset, true);
-  EVT VT = isPPC64 ? MVT::i64 : MVT::i32;
+  EVT VT = IsPPC64 ? MVT::i64 : MVT::i32;
```
Comment (amy-kwan) [inline llvm/lib/Target/PowerPC/PPCISelLowering.cpp:5252]: Is the reason why we don't use `Subtarget.getScalarIntVT()` here because we have `IsPPC64`, so if we don't use it in the function, the argument becomes unused?

File: llvm/lib/Target/PowerPC/PPCSubtarget.h [hunk 2]
```diff
@@ -216,8 +216,8 @@ class PPCSubtarget : public PPCGenSubtargetInfo {
   bool isSVR4ABI() const { return !isAIXABI(); }
   bool isELFv2ABI() const;
 
-  bool is64BitELFABI() const { return  isSVR4ABI() && isPPC64(); }
-  bool is32BitELFABI() const { return  isSVR4ABI() && !isPPC64(); }
+  bool is64BitELFABI() const { return isSVR4ABI() && isPPC64(); }
```
Comment (amy-kwan) [inline llvm/lib/Target/PowerPC/PPCSubtarget.h:219]: Unrelated changes?

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp (hunk 1 above)
Comment (lei137) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:5252]: It is cause this is a static function that doesn't have access to `Subtarget`.  This function is also called from another static function that has no access to `Subtarget` and uses `isPPC64` in a different way.  I didn't think this patch should include trying to fix those dependencies.  I updated the var since it didn't follow the variable name guideline where variables should start with a capital.

File: llvm/lib/Target/PowerPC/PPCSubtarget.h (hunk 2 above)
Comment (lei137) [reply llvm/lib/Target/PowerPC/PPCSubtarget.h:219]: Yes this is.  It was auto updated by clang-format.  Not sure why it was identified as needing change but I left it since its a valid format update.

File: llvm/lib/Target/PowerPC/PPCISelLowering.cpp (hunk 1 above)
Comment (amy-kwan) [reply llvm/lib/Target/PowerPC/PPCISelLowering.cpp:5252]: Yeah, I agree that this patch should not fix those dependencies. Thanks for explaining.

**Review by amy-kwan — APPROVED** [review]
LGTM.
